web/src/main/java/org/springframework/security/web/server/authentication/logout/HeaderWriterServerLogoutHandler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.server.authentication.logout;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.server.WebFilterExchange;
+import org.springframework.security.web.server.header.ServerHttpHeadersWriter;
+import org.springframework.util.Assert;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * <p>A {@link ServerLogoutHandler} implementation which writes HTTP headers during logout.</p>
+ *
+ * @author MD Sayem Ahmed
+ * @since 5.2
+ */
+public final class HeaderWriterServerLogoutHandler implements ServerLogoutHandler {
+	private final ServerHttpHeadersWriter headersWriter;
+
+	/**
+	 * <p>Constructs a new instance using the {@link ServerHttpHeadersWriter} implementation.</p>
+
+	 * @param headersWriter a {@link ServerHttpHeadersWriter} implementation
+	 * @throws IllegalArgumentException if the argument is null
+	 */
+	public HeaderWriterServerLogoutHandler(ServerHttpHeadersWriter headersWriter) {
+		Assert.notNull(headersWriter, "ServerHttpHeadersWriter must not be null");
```

Comment:
Curious if we should keep it `consistent` with the [servlet version](https://github.com/spring-projects/spring-security/blob/4a6e91531afb41392a44b9eb03d0636505a486fb/web/src/main/java/org/springframework/security/web/authentication/logout/HeaderWriterLogoutHandler.java#L41) which is `headersWriter cannot be null`   🤔

Reply:
Hey thanks for the review. Yeah it makes sense to keep it consistent across both version, at least the style of the message. I've forced-pushed the consistent version.